github/repos.go

```diff
@@ -767,6 +767,47 @@ type Protection struct {
 	RequiredConversationResolution *RequiredConversationResolution `json:"required_conversation_resolution"`
 }
 
+type ProtectionRule struct {
```

Comment:
Apparently there is already `ProtectionRule` already defined here - 

https://github.com/google/go-github/blob/7a077bbd7313615d4f924f12b065913ee94e1d58/github/repos_environments.go#L52

May be I will name as `BranchProtectionRule` here